core/src/main/java/org/bitcoinj/base/internal/ByteArray.java

```diff
@@ -0,0 +1,54 @@
+/*
```

Comment:
Do we already have a second use (besides Bech32Bytes) for this? If not, we could start by folding this into Bech32Bytes – especially if we choose to extract the "first byte + rest" into a SegwitBech32Bytes class (see comment above).

Reply:
> Do we already have a second use?

`AesKey` is the one that was on my mind, and we're considering creating a similar wrapper for `ECKey`.

I think there are probably plenty of other cases where having a typed, immutable `byte[]`-wrapper class would be handy. It could also have some utility methods like `formatHex` that are currently in `ByteUtils`.